src/app/ApplicationTemplate.Tests/Business/PostServiceShould.cs

```diff
@@ -20,22 +18,42 @@ public async Task It_Should_GetAll()
 			results.Should().NotBeNullOrEmpty();
 		}
 
-		[Fact]
-		public async Task It_Should_GetOne()
+		[Theory]
+		// Valid cases
+		[InlineAutoData(1)]
+		// Invalid cases
+		[InlineAutoData(-1)]
+		[InlineAutoData(0)]
+		[InlineAutoData(int.MaxValue)] // This will be invalid most of the time but it can be valid if the max amount of posts if created.
+		public async Task GetPostOrThrowException(int givenId)
 		{
 			// Arrange
-			var postId = 1;
+			var posts = await SUT.GetPosts(DefaultCancellationToken);
+			var postsNumber = posts.Count;
 
 			// Act
-			var result = await SUT.GetPost(DefaultCancellationToken, postId);
+			Func<Task<PostData>> act = () => SUT.GetPost(DefaultCancellationToken, givenId);
 
-			// Assert
-			result.Should().NotBeNull();
-			result.Id.Should().Be(postId);
+			if (givenId >= 1 && givenId <= postsNumber)
```

Comment:
isnt it weird to have ifs in the test? does it mean it should be two tests instead?